addons/test_mail/models/test_mail_corner_case_models.py

```diff
@@ -184,3 +185,29 @@ class MailTestNotMailThread(models.Model):
     name = fields.Char()
     company_id = fields.Many2one('res.company')
     customer_id = fields.Many2one('res.partner')
+
+class MailTestActivityInSystray(models.Model):
+    """Simple model used to plan activities to be displayed in the systray"""
+    _name = "mail.test.activity.in.systray"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+    _description = "Test activities in systray"
+
+    name = fields.Char()
+    category = fields.Selection(selection=[('cat1', 'Category 1'), ('cat2', 'Category 2')])
+
+    @api.model
+    def _sort_activities_by_record_by_module(self, activities):
+        """ Override to separate model related activities between module_1
+            and module_2 modules depending on the value of their category field.
+        """
+        activities_by_rec_id_by_module = defaultdict(lambda: defaultdict(lambda: self.env["mail.activity"]))
+        group_custom_info = defaultdict(lambda: defaultdict(lambda: False))
+        group_custom_info['mail'] = {'name': _('Category 1'), 'domain': [['category', '=', 'cat1']]}
+        group_custom_info['test_mail'] = {'name': _('Category 2'), 'domain': [['category', '=', 'cat2']]}
+        for activity in activities:
+            record = self.env[activity.res_model].browse(activity.res_id)
+            if record.category == 'cat1':
+                activities_by_rec_id_by_module['mail'][activity.res_id] += activity
+            else:
+                activities_by_rec_id_by_module['test_mail'][activity.res_id] += activity
+        return activities_by_rec_id_by_module, group_custom_info
```

Comment:
Not a python expert, but that's honestly not a good way to test code.
One of the core function you add in the diff should be tested, but if you completely override in the test environment, this kills the purpose of testing the actual code.

If you really need to do some patching in test, please keep it very minimal. None is even better if you can :)

Reply:
> Not a python expert, but that's honestly not a good way to test code. One of the core function you add in the diff should be tested, but if you completely overwrite it in the test environment, this kills the purpose of testing the actual code.
> 
> If you really need to do some patching in test, please keep it very minimal. None is even better if you can :)

I get your point. The idea here was not really to test the override of `_sort_activities_by_record_by_module` in mass_mailing_sms as it is already done in the fix PR. The purpose here was more to test that it was indeed possible to split activities in two groups with an override. But it's a bit useless I guess if we already have the test in mass_mailing_sms (once the fix pr is merged).

So I propose to remove this test.